specs/NetworkSpec/NetworkDocumentation.md

```diff
@@ -0,0 +1,65 @@
+# 1. Overview
+Near Protocol uses its own implementation of a custom peer-to-peer network
+Peers who join network are represented by nodes and connected between them by edges.
+
+The purpose of this document is to describe inner workings of `near-network` package;
+and to be used as reference by future engineers to understand network code without any prior knowledge.
+
+# 2. Code structure
+`near-network` runs on top of `actor` framework called `Actix` (https://actix.rs/docs/).
+Code structure is split between 4 actors `PeerManagerActor`, `PeerActor`, `RoutingTableActor`, `EdgeValidatorActor`
```

Comment:
Code structure belongs to specific implementation of the protocol, not the protocol itself and therefore should not appear in this repository

Reply:
@bowenwang1996 In our quarterly goal, we wrote that we want to have a reference level documentation for `near-network`. If this is not the right place for such documentation, what is the right place?

Reply:
For posterity: we've decided to put that in https://github.com/near/nearcore/tree/master/docs